### PR TITLE
fix: enable component client for integration tests

### PR DESCRIPTION
**Pull request by aludwiko**
References #1743

### CODE REVIEW

**Review by efgpinto — APPROVED** [review]
LGTM.. I think we might need another utility method as commented below but that can be address in a follow-up.

File: sdk/java-sdk-spring-testkit/src/main/java/kalix/spring/testkit/KalixIntegrationTestKitSupport.java [hunk 1]
```diff
@@ -44,11 +51,32 @@ public abstract class KalixIntegrationTestKitSupport {
...
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Utility method to execute a deferred call and wait for the response.
+   */
+  protected <T> T execute(DeferredCall<Any, T> deferredCall) {
+    return execute(deferredCall, timeout);
+  }
```
[the remark this answers is not in the document]
Comment (aludwiko) [reply sdk/java-sdk-spring-testkit/src/main/java/kalix/spring/testkit/KalixIntegrationTestKitSupport.java:81]: actually, the solution for this problem is a bit different: https://github.com/aludwiko/kalix-saga-patterns/blob/main/src/main/java/com/example/cinema/application/Response.java#L10

[the remark this answers is not in the document]
Comment (efgpinto) [reply sdk/java-sdk-spring-testkit/src/main/java/kalix/spring/testkit/KalixIntegrationTestKitSupport.java:81]: Aha, I see. fair enough 👍

File: samples/java-spring-eventsourced-customer-registry/src/it/java/customer/api/CustomerIntegrationTest.java [hunk 2]
```diff
@@ -27,26 +26,24 @@
 @SpringBootTest(classes = Main.class)
 public class CustomerIntegrationTest extends KalixIntegrationTestKitSupport {
 
-
   @Autowired
   private WebClient webClient;
 
+  @Autowired
+  private ComponentClient componentClient;
```
Comment (octonato) [inline samples/java-spring-eventsourced-customer-registry/src/it/java/customer/api/CustomerIntegrationTest.java:33]: does that mean that we can wire the `ComponentClient` anywhere? Like wire it into an Entity?

File: sdk/java-sdk-spring-testkit/src/main/java/kalix/spring/testkit/KalixIntegrationTestKitSupport.java [hunk 3]
```diff
@@ -44,11 +51,32 @@ public abstract class KalixIntegrationTestKitSupport {
...
   @AfterAll
   public void afterAll() {
     logger.info("Stopping Kalix TestKit...");
     kalixTestKit.stop();
   }
+
+  /**
+   * Utility method to execute a deferred call and wait for the response for a specified time.
+   */
+  protected <T> T execute(DeferredCall<Any, T> deferredCall, Duration timeout) {
```
[the remark this answers is not in the document]
Comment (aludwiko) [reply sdk/java-sdk-spring-testkit/src/main/java/kalix/spring/testkit/KalixIntegrationTestKitSupport.java:68]: I'm not sure about this method as well, my motivation was that such a method will appear in any integration test, but maybe it's fine and better for the user to decide when/if to add it.

[the remark this answers is not in the document]
Comment (octonato) [reply sdk/java-sdk-spring-testkit/src/main/java/kalix/spring/testkit/KalixIntegrationTestKitSupport.java:68]: I would remove it. It's just a thin layer on top of a very common CompletionStage operation. We don't need to abstract this away for users.

File: samples/java-spring-eventsourced-customer-registry/src/it/java/customer/api/CustomerIntegrationTest.java (hunk 2 above)
Comment (aludwiko) [reply samples/java-spring-eventsourced-customer-registry/src/it/java/customer/api/CustomerIntegrationTest.java]: No, it's not possible, nothing has changed.

**Pull request merged**